### PR TITLE
Adds ipv4 format checker for jsonschema

### PR DESCRIPTION
**Pull request by qduk**

### CODE REVIEW

**Comment by qduk** [comment]
As shown in issue https://github.com/networktocode/schema-enforcer/issues/94 I could have imported the `draft7_format_checker`. I just opted to do this one as I'm not sure if we need all of the format checkers or not. If you guys just want that one imported instead, I can write tests for each of those. Just let me know!

**Review by PhillSimonds — APPROVED** [review]
LGTM! One comment inline, although it's just a nit. Ty for putting this together!

**Comment by dgarros** [comment]
Thanks for the fix @qduk, personally I think it would be better to check for all formats in `draft7_format_checker`

**Comment by qduk** [comment]
@dgarros @PhillSimonds I will update this PR to have tests for all the validations checked in the `draft7_format_checker`.

File: tests/test_jsonschema.py [hunk 1]
```diff
@@ -94,6 +100,19 @@ def test_validate(schema_instance, valid_instance_data, invalid_instance_data, s
             == "Additional properties are not allowed ('fun_extr_attribute' was unexpected)"
         )
 
+    @staticmethod
+    def test_ipv4_format_checker(schema_instance, invalid_format_instance_data):
+        """Test ipv4 format checker
+
+        Args:
+            schema_instance (JsonSchema): Instance of JsonSchema class
+        """
```
[the remark this answers is not in the document]
Comment (qduk) [reply tests/test_jsonschema.py]: I took out the args from the docstring. Thanks for letting me know!

**Comment by PhillSimonds** [comment]
LGTM!

**Review by itdependsnetworks — APPROVED** [review]

File: schema_enforcer/schemas/jsonschema.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ def __init__(self, schema, filename, root):
         ]
         self.validator = None
         self.strict_validator = None
+        self.format_checker = draft7_format_checker
```
Comment (dgarros) [inline schema_enforcer/schemas/jsonschema.py:35]: please can you move it outside of init, at the root of the class, since its value is always the same.

**Review by dgarros — COMMENTED** [review]
Thanks for making these changes and adding the tests
couple more comments from me but we're almost there, 
thanks

File: pyproject.toml [hunk 3]
```diff
@@ -25,6 +25,10 @@ jsonref = "^0.2"
 pydantic = "^1.6.1"
 rich = "^9.5.1"
 ansible = "^2.8.0"
+rfc3987 = "^1.3.8"
```
[the remark this answers is not in the document]
Comment (jmcgill298) [reply pyproject.toml:28]: In the past, these dependencies were not provided as part of jsonschema isntall, if you want to use these format checkers, then you would need to install them separately.

File: schema_enforcer/schemas/jsonschema.py (hunk 2 above)
Comment (jmcgill298) [reply schema_enforcer/schemas/jsonschema.py:35]: I don't think this is a good idea; if someone wanted to monkey patch and use a different draft, then that would change it for every instance.

Comment (dgarros) [reply schema_enforcer/schemas/jsonschema.py:35]: I'm not gonna argue this one but if someone need to monkey patch this class we should revisit the design altogether.

**Review by dgarros — APPROVED** [review]

**Comment by dgarros** [comment]
@qduk please can you fix the conflicts  and we should be good to merge

**Comment by qduk** [comment]
@dgarros @PhillSimonds Conflicts resolved. Thanks guys!

**Pull request merged**